### PR TITLE
Add r-bigsnpr from CRAN using conda_r_skeleton_helper

### PR DESCRIPTION
**Pull request by pettyalex**
This package depends on several packages that are not yet present in conda-forge, including r-bigsparser, r-bigutilsr, and r-runonce. I've made PRs to add each of those individally:

https://github.com/conda-forge/staged-recipes/pull/24240
https://github.com/conda-forge/staged-recipes/pull/24239
https://github.com/conda-forge/staged-recipes/pull/24238

Hopefully all three of these can be added, at which point r-bigsnpr can be added to conda-forge. I will be working through the checklists for all four of these PRs. Each of the four packages recipes was generated by the helper script https://github.com/bgruening/conda_r_skeleton_helper as directed in the conda-forge contribution guidelines.

Edit: I just noticed that conda-forge policies allow multiple packages in a single PR. I could move all four into this PR if preferred. Given that I've already made the 4 PRs and have the three dependencies passing CI checks, I'm also OK waiting for them to go in before merging this.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by pettyalex** [comment]
Once all three dependencies have gone in, this should be buildable.

**Comment by mfansler** [comment]
This package wraps some PLINK functionality, which is available on the Bioconda channel. Would it make more sense for this package to be on Bioconda instead of Conda Forge?

**Comment by pettyalex** [comment]
Good call. I started off adding all of these to Bioconda, saw their guidance that if things from CRAN can easily be packaged for conda-forge instead that is preferred. 

I didn't think about `plink` because it's not a dependency on CRAN, and not required for this package to work. I know that anecdotal evidence is weak, but my research group hasn't ever used the plink-wrapping functions and I wasn't familiar with them. This package also seems to wrap `beagle` as well, hrm: https://privefl.github.io/bigsnpr/reference/snp_beagleImpute.html
https://privefl.github.io/bigsnpr/reference/download_beagle.html

Given that these are very optional, and I don't see a way to configure the package to easily use conda-installed versions of plink or beagle anyway, I'd prefer to see this in conda-forge with its wider platform support if possible. That said, if you or any other conda-forge maintainers feel that this belongs in bioconda I can easily move it over there.

**Comment by mfansler** [comment]
Okay. If "*very optional*", then lets just keep it here. I concur that there doesn't appear to be option-like specifications for the PLINK and beagle PATHs, so adding them as dependencies wouldn't automate connecting them.

I just figured I'd ask since this is specifically a key advantage over CRAN - that we can additionally guarantee non-R libraries.

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for submitting this! Review notes:

- linked libraries need to be added
- win build needs https://github.com/conda-forge/r-runonce-feedstock/pull/2
- run testthat tests (suggestion)

**Comment by pettyalex** [comment]
@mfansler thanks so much for all of the fixups. Really appreciated.

This may not be the specific place to ask general questions, but is there a way for recipes to specifically request the MKL implementations of BLAS if they're available, without requiring it? We've got a whole host of tools that run much better with the MKL but when installed with default settings from conda-forge they get OpenBLAS.

**Comment by pettyalex** [comment]
Hmm, it looks like it uses https://anaconda.org/conda-forge/quadprog for its tests?

File: recipes/r-bigsnpr/meta.yaml
Comment (mfansler) [inline recipes/r-bigsnpr/meta.yaml:43]: If these are not in `Depends` nor `Imports` then the intent of the author(s) is presumed to be that they are optional. Since a primary use of Conda packages is for minimal deployments (e.g., a specific step in a processing pipeline) we don't want to add on unnecessary components to installation. Let's move all these to only `test` requirements.

```suggestion
```

File: recipes/r-bigsnpr/meta.yaml
Comment (mfansler) [inline recipes/r-bigsnpr/meta.yaml]: Move to `test` only.

```suggestion
    - r-rmio
    - r-roptim >=0.1.6
```

File: recipes/r-bigsnpr/meta.yaml
Comment (mfansler) [inline recipes/r-bigsnpr/meta.yaml:66]: Move to `test` only.

```suggestion
```

File: recipes/r-bigsnpr/meta.yaml
Comment (mfansler) [inline recipes/r-bigsnpr/meta.yaml:74]: Move to `test` only.

```suggestion
```

File: recipes/r-bigsnpr/meta.yaml
Comment (mfansler) [inline recipes/r-bigsnpr/meta.yaml]: Move to `test` only.

```suggestion
    - r-rmio
    - r-roptim >=0.1.6
    - r-runonce >=0.2.3
    - r-vctrs
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Still waiting for an answer on how to deal with the OSX missing `libcxx` error. In the meantime, please implement these changes, which should get the Windows builds working and minimizes the dependencies.

File: recipes/r-bigsnpr/meta.yaml
Comment (mfansler) [inline recipes/r-bigsnpr/meta.yaml:104]: Moving to `test`. Also adding `r-pcadapt`, since that is now available.

```suggestion
    - r-testthat
    - r-dbplyr
    - r-dplyr
    - r-fdrtool
    - r-gaston
    - r-hmisc
    - r-mutil
    - r-pcadapt
    - r-quadprog
    - r-rsqlite
    - r-xgboost
```

File: recipes/r-bigsnpr/meta.yaml
Comment (mfansler) [inline recipes/r-bigsnpr/meta.yaml:98]:
```suggestion
    - r-rmio
    - r-roptim >=0.1.6
    - r-runonce >=0.2.3
    - r-vctrs
```

File: recipes/r-bigsnpr/meta.yaml
Comment (mfansler) [inline recipes/r-bigsnpr/meta.yaml:66]: Move to `test` only:

```suggestion
    - r-rmio
    - r-roptim >=0.1.6
    - r-runonce >=0.2.3
    - r-vctrs
```

**Comment by mfansler** [comment]
> This may not be the specific place to ask general questions, but is there a way for recipes to specifically request the MKL implementations of BLAS if they're available, without requiring it? We've got a whole host of tools that run much better with the MKL but when installed with default settings from conda-forge they get OpenBLAS.

Most of the time<sup>**1**</sup> this should be left to the end user, not something to be specified in the recipe. Using `libblas` and `liblapack` in the recipe will build against the generic reference implementation (netlib). This is then flexible to run atop any of the implementations. MKL is great for Intel x86_64 users, but Apple Silicon users highly benefit from the Accelerate and AMD users might prefer OpenBLAS.

For end users to specify MKL in an environment, add the specification:

**YAML**
```yaml
  - blas=*=mkl
```

or, with the **CLI**
```bash
conda install 'blas=*=mkl'
```

[This StackOverflow answer](https://stackoverflow.com/a/70241376/570918) has some related content (though focus is Python's NumPy).

---

<sub>**[1]**: The exception here is if the package actually will use library-specific functions at compile-time. In that case one can use variants to build against different implementations. However, this is rare and does not sound like the situation here.</sub>

File: recipes/r-bigsnpr/meta.yaml
Comment (mfansler) [inline recipes/r-bigsnpr/meta.yaml:96]: Should be:

```suggestion
    - r-rmutil
```

**Pull request merged**
